odoo/addons/base/tests/test_expression.py

```diff
@@ -2698,3 +2698,25 @@ def test_not_or_multiple_one2many_field_with_subfield(self):
                     ('name', '=', 'Daniel'),
             ])
         ])
+
+
+class TestOptimize(TransactionCase):
+
+    def assert_optimized_domain(self, env, domain, expected, full_opti=False):
+        model = env['res.partner']
+        optimized = Domain(domain).optimize(model, full=full_opti)
+        expected = Domain(expected).map_conditions(lambda c: c.optimize(model, full=full_opti))
+        self.assertEqual(optimized, expected, f"\nFor initial domain: {domain}\nBecame: {optimized}")
+
+    def test_bypass_comodel_id_lookup(self):
+        expected_domain_by_su = {
+            True: [('company_id', 'in', [2])],
+            False: [('company_id', 'any', [('id', 'in', [2])])]
+        }
+        for sudo, expected_domain in expected_domain_by_su.items():
+            self.assert_optimized_domain(
```

Comment:
You can inline this method here.

Create the domains using `Domain` object for what you expect and you can then just compare using something like 
```
assertEqual(Domain('company_id.id', '=', 2).optimize(model.sudo(su=sudo), full=True), expected_domain)
```